app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,28 @@ def index
     @user = User.find(params[:user_id])
     @posts = @user.posts.order(created_at: :desc)
   end
+
+  def new
+    @current = current_user
+  end
+
+  def create
+    created_post = current_user.posts.build(post_params)
+
+    respond_to do |format|
+      format.html do
+        if created_post.save
+          redirect_to user_post_path(created_post.author_id, created_post.id), notice: 'Post created !'
+        else
+          render :new, alert: 'Post not created, please try again!'
+        end
+      end
+    end
+  end
+
```

Comment:
- [ ] Kindly note that there is no column in your post table called author_id instead the user is referenced in the post table by the `user_id` column. Kindly fix this by either updating your post table(migration) to reference the user by the `post_id` or user the `user_id in your controllers and views.

![image](https://user-images.githubusercontent.com/65068771/161387720-a7624a32-0963-4cab-9bb6-b45260c10113.png)

- [ ] Check your modal also if it has the correct association.